### PR TITLE
Write the CI Signal report to a file & markdown formatting

### LINKED ISSUE #2358: ci-signal-reporter, Write report to file
#### Write report to file
:green_circle: ci-signal-report-cli

**Summary:** The report output is getting written to a local file which can be shared instead of plain text.

<details>
<summary>Current situation</summary>
Currently the ci-reporter CLI only print the report to the console.
</details>

<details>
<summary>Desired outcome</summary>
The ci reporter CLI should be able to create a new local file to write the report to.

This probably goes along with defining a new flag `-f` to specify the file name and format.
</details>
<br>

**Benefits:** This would simplify how reports are versioned, shared and managed.

**Concerns:** None

### LINKED ISSUE #2359: ci-signal-reporter, Markdown formatted print out
#### Markdown formatted print out

:green_circle: ci-signal-report-cli

**Summary:** Markdown used to highlight report printing

<details>
<summary>Current situation</summary>
Currently the report is getting printed out in plain text format without any highlighting.
</details>

<details>
<summary>Desired outcome</summary>
Format report print out to highlight it more and shorten it in size (named links)

A new flag should be added to control this behavior
</details>
<br>

**Benefits:** Improve report visibility and make it more compact

**Concerns:** None

### PR DESCRIPTION
**Pull request by leonardpahlke**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add new & requested features to the CI Signal report CLI.
* Write report to a file
* report table output get's written in markdown format 

#### Which issue(s) this PR fixes:

Fixes #2359
Fixes #2358

#### Special notes for your reviewer:

I added a small new dependency to visualize the load time for the data request.
This gives feedback to the user that everything is ok, as the data query can take a few seconds (depending on the request). (see [go-spin](https://github.com/tj/go-spin))

#### Does this PR introduce a user-facing change?

```release-note
Add the flag 'flag' / 'f' command to the CLI, which writes the report output to a local file.
Report table output now has Markdown formatting
```

cc @kubernetes/ci-signal 

/label tide/merge-method-squash
/assign @puerco

### CODE REVIEW

File: cmd/ci-reporter/cmd/root.go [hunk 1]
```diff
@@ -178,19 +191,50 @@ func (r *CIReporters) CollectReportDataFromReporters(cfg *Config) (*CIReportData
...
 		// print report in json format
 		d, err := reports.Marshal()
 		if err != nil {
 			return nil
 		}
-		fmt.Print(string(d))
+		_, err = out.WriteString(string(d))
+		if err != nil {
+			return errors.Wrap(err, "could not write to output stream")
+		}
```
Comment (puerco) [inline cmd/ci-reporter/cmd/root.go:227]: Keep this if block but then return early and remove the else block bellow to simplify

File: cmd/ci-reporter/cmd/root.go [hunk 2]
```diff
@@ -205,25 +249,26 @@ func PrintReporterData(cfg *Config, reports *CIReportDataFields) error {
...
-				categoryCounts += fmt.Sprintf("%s:%d\n", category, categoryCount)
+				categoryCounts += fmt.Sprintf("%s:%d ", category, categoryCount)
 			}
-			if cfg.ShortReport {
-				table.SetFooter([]string{fmt.Sprintf("Total: %d", len(data)), "", categoryCounts, ""})
-			} else {
-				table.SetFooter([]string{fmt.Sprintf("Total: %d", len(data)), "", categoryCounts, "", "", "", ""})
+			_, err = out.WriteString(fmt.Sprintf("\nSUMMARY - Total:%d %s\n", len(data), categoryCounts))
+			if err != nil {
+				return errors.Wrap(err, "could not write to output stream")
```
Comment (puerco) [inline cmd/ci-reporter/cmd/root.go:270]: When we're dealing with a result with a single error, we usually prefer:

```suggestion
			if _, err := out.WriteString(fmt.Sprintf("\nSUMMARY - Total:%d %s\n", len(data), categoryCounts)); err != nil {
				return errors.Wrap(err, "could not write to output stream")
```

**Review by puerco — COMMENTED** [review]
Nice, thank you @leonardpahlke Just two nits but it looks fine to me:

File: cmd/ci-reporter/cmd/root.go (hunk 1 above)
Comment (leonardpahlke) [reply cmd/ci-reporter/cmd/root.go:227]: Okay, sounds good! I also fixed line 222 to return the error if marshalling failed

**Comment by leonardpahlke** [comment]
Sounds good! Thank you 😃

**Pull request merged**